### DISCUSSION TITLE
Need some help with the tag-xml to create my maps please

### DISCUSSION THREAD

**Discussion by utack** [mapsforge/mapsforge#1514] [Q&A] [answered]
Hello folks,

I have tried to read the docu over at http://mapsforge.org/tag-mapping https://raw.githubusercontent.com/mapsforge/mapsforge/master/resources/tag-mapping.xsd but unfortunately need some help.
It would be much appreciated!

First of all I wanted to use a wildcard to add all buildings to my map, but it is not working

```
	<ways>
		<osm-tag key="building" value="yes" equivalent-values="%s" zoom-appear="16"/>
	</ways>
```

I was hoping this would map any building type to building=yes

Second of all I wanted to map some things into one
for example landuse=forest and natural=wood should idealy be mapped into one of these, do not care which one, since I render them the same and want to to recuce map footprint.

would I need a tag-group or tag-mapping for this?
I have tried both but with no success.

Thank you and best
Jakob

**Comment by devemux86** [comment, marked as answer]
>  I wanted to use a wildcard to add all buildings to my map

- See the [documentation](https://github.com/mapsforge/mapsforge/blob/master/resources/tag-mapping.xsd#L44-L49) for wildcards and search the tag-mapping for "%f" and "%s" to see their use.
We don't use the"%s" in tag-mapping as it is recommended to declare the specific string values.

- Just include the needed tags and use the same render rule for them, see an example in [default theme](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-themes/src/main/resources/assets/mapsforge/default.xml#L40).

**Reply by utack** [reply]
Thank you so much

I have no problem with this, but the first 15 pages of taginfo gives me 550+ values for building and that makes the xml a bit large, it was not entirely clear to make the wildcard is only for the rendering!

The second point is good to know, I will do it like this!

**Reply by devemux86** [reply]
See how we declare the buildings in the tag-mapping.
(we use the "%f" but not the "%s")

**Reply by utack** [reply]
Sorry I am still confused
You are not using it due to performance overhead, but shouldn't it technically work to use my idea

```
	<ways>
		<osm-tag key="building" value="yes" equivalent-values="%s" zoom-appear="16"/>
	</ways>
```

This leads to all tags other than "yes" being discarded
Do i need a quantity specifier that allows there to be an arbitrary number of %s?

**Reply by devemux86** [reply]
You need to use correct syntax, please see the tag-mapping for an [example](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-writer/src/main/config/tag-mapping.xml#L505C13-L505C76):
```xml
<osm-tag key="building:colour" renderable="false" value="%s" />
```
